README.md

```diff
@@ -76,7 +79,8 @@ To delete the cluster and the infrastructure created earlier, run:
 hypershift destroy cluster \
   --aws-creds ~/.aws/credentials \
   --namespace clusters \
-  --name example
+  --name example \
+  --base-domain hypershift.example.com
```

Comment:
Why do we need to re-specify this on destroy? Seems like a UX regression to me

Reply:
If the HC is deleted independent the infra, then we have no way to look up the `base-domain` of a cluster of a particular name right?

Reply:
I agree, but isn't the typical case with this flow that we do in fact have the HostedCluster? In which case, we could use the discovered value by default and use the flag only if the HostedCluster isn't available.